app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt

```diff
@@ -1,38 +1,153 @@
 package com.github.sdp_begreen.begreen
 
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.util.Log
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
 import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.ValueEventListener
 import com.google.firebase.database.ktx.database
 import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.tasks.await
+import java.io.ByteArrayOutputStream
 import java.util.concurrent.CompletableFuture
 
 /**
- * Database implementation using Firebase
+ * Database implementation using Firebase's realtime database
  */
-class FirebaseDB : Database() {
+class FirebaseDB() {
 
-    private val firebaseDB: DatabaseReference = Firebase.database.reference
+    /**
+     * Singleton value to access the database
+     */
+    companion object {
+        val db: FirebaseDB = FirebaseDB()
+    }
+
+    /**
+     * Overwritten during tests! (to perform tests locally)
+     */
+    var databaseReference: DatabaseReference = Firebase.database.reference

```

Comment:
As we are using the emulator for test, I think it is unnecessary to keep the possibility to override the reference during test, so I would change the reference to a `private val`